### PR TITLE
Fix user deletion

### LINKED ISSUE #45: Unconfirmed user deletion seems broken
### Description - What happened? *

Unconfirmed user deletion seems to break under certain circumstances and leaves remainders in the DB, which results in a failing "Users & Roles" download.

### Expected Behavior *

Users are cleaned up correctly.

### Observed Behavior *

DB deletion fails with the log output below.

### Steps to Reproduce

_No response_

### Context Information

_No response_

### Relevant log output

```shell
2024-03-12 09:41:58,064 ERROR [io.qua.sch.com.run.StatusEmitterInvoker] (executor-thread-8358) Error occurred while executing task for trigger IntervalTrigger [id=1_de.sovity.authorityportal.web.pages.usermanagement.UnconfirmedUserDeletionService#deleteUnconfirmedUsersAndOrganizations, interval=900000]: java.util.concurrent.CompletionException: org.jooq.exception.IntegrityConstraintViolationException: SQL [delete from "organization" where "organization"."mds_id" in (?, ?, ?)]; ERROR: update or delete on table "organization" violates foreign key constraint "fk_user_organization_id" on table "user"
```

### Screenshots

_No response_

### PR DESCRIPTION
**Pull request by jridderbusch**
closes #45 

```[tasklist]
### Checklist
- [x] The PR title is short and expressive.
- [x] I have updated the CHANGELOG.md. See [changelog_update.md](https://github.com/sovity/authority-portal/tree/main/docs/dev/changelog_updates.md) for more information.
- [x] I have updated the Deployment Migration Notes Section in the CHANGELOG.md for any configuration / external API changes.
- [x] I have performed a **self-review**
```

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -29,6 +29,7 @@ please see [changelog_updates.md](docs/dev/changelog_updates.md).
 - Fixed an issue where upon clicking "back to application" on the logout confirmation page, the user would appear to be logged out instead of the logout being canceled ([#8](https://github.com/sovity/authority-portal/issues/8))
 - Fixed redirection to broker catalog with participant's data offers pre-filtered ([#44](https://github.com/sovity/authority-portal/issues/44))
 - Fixed an issue where an error 403 would be displayed under certain circumstances ([#107](https://github.com/sovity/authority-portal/issues/107))
+- Fixed deletion process, if a whole organization gets deleted ([#45](https://github.com/sovity/authority-portal/issues/45))
```
Comment (kamilczaja) [inline CHANGELOG.md:32]: wording nit: "Fixed an issue where deleting the last user of an organization would delete the organization but leave the user in an invalid state"

File: authority-portal-backend/authority-portal-db/src/main/resources/db/testdata/V1_1__Initial_Schema_Test_Data.sql [hunk 2]
```diff
@@ -24,8 +24,6 @@ values ('00000000-0000-0000-0000-000000000007', null, 'APPROVED');
 insert into "user" (id, organization_mds_id, registration_status)
 values ('00000000-0000-0000-0000-000000000008', null, 'PENDING');
 insert into "user" (id, organization_mds_id, registration_status)
-values ('00000000-0000-0000-0000-000000000009', null, 'PENDING');
-insert into "user" (id, organization_mds_id, registration_status)
```
Comment (kamilczaja) [inline authority-portal-backend/authority-portal-db/src/main/resources/db/testdata/V1_1__Initial_Schema_Test_Data.sql:28]: What does this change do?

**Review by kamilczaja — CHANGES_REQUESTED** [review]

File: authority-portal-backend/authority-portal-db/src/main/resources/db/testdata/V1_1__Initial_Schema_Test_Data.sql (hunk 2 above)
Comment (jridderbusch) [reply authority-portal-backend/authority-portal-db/src/main/resources/db/testdata/V1_1__Initial_Schema_Test_Data.sql:28]: Funny you should ask... I think this was about a user in the DB that has no mdsId, which is an invalid state as of now.

**Review by kamilczaja — APPROVED** [review]
LGTM

**Pull request merged**